arelle/ModelXbrl.py

```diff
@@ -379,69 +311,61 @@ def closeViews(self):
                     self.views[0].close()
 
     @property
-    def displayUri(self):
+    def displayUri(self) -> Any:
         if hasattr(self, "ixdsDocUrls"):
             return "IXDS {}".format(", ".join(os.path.basename(url) for url in self.ixdsDocUrls))
         elif hasattr(self, "uri"):
             return self.uri
         else:
             return self.fileSource.url
 
-    def relationshipSet(self, arcrole, linkrole=None, linkqname=None, arcqname=None, includeProhibits=False) -> ModelRelationshipSetClass:
+    def relationshipSet(self, arcrole: str, linkrole: tuple[str] | str | None = None, linkqname: QName | None = None, arcqname: QName | None = None, includeProhibits: bool = False) -> ModelRelationshipSetClass:
         """Returns a relationship set matching specified parameters (only arcrole is required).
 
         Resolve and determine relationship set.  If a relationship set of the same parameters was previously resolved, it is returned from a cache.
 
         :param arcrole: Required arcrole, or special collective arcroles 'XBRL-dimensions', 'XBRL-formula', and 'Table-rendering'
-        :type arcrole: str
         :param linkrole: Linkrole (wild if None)
-        :type linkrole: str
         :param arcqname: Arc element qname (wild if None)
-        :type arcqname: QName
         :param includeProhibits: True to include prohibiting arc elements as relationships
-        :type includeProhibits: bool
-        :returns: [ModelRelationship] -- Ordered list of effective relationship objects per parameters
         """
         from arelle import ModelRelationshipSet
-        key = (arcrole, linkrole, linkqname, arcqname, includeProhibits)
+        key: tuple[str, tuple[str] | str | None, QName | None, QName | None, bool] = (arcrole, linkrole, linkqname, arcqname, includeProhibits)
```

Comment:
`arcrole`, `linkrole`, etc are all already types, so the type of `key` can be inferred:
```suggestion
        key = (arcrole, linkrole, linkqname, arcqname, includeProhibits)
```